### ISSUE TITLE
test_payout_e2e failing occassionally

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#4219]
It seems that there is a loop in the construction of the e2e circuit, causing one of the four nodes in the tests to miss the payout, and to trigger the assert.

```
======================================================================
FAIL: Check if payouts work for an e2e-linked circuit
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/Users/martijndevos/Documents/tribler/Tribler/Test/Community/Tunnel/test_community.py", line 295, in test_payouts_e2e
    self.assertTrue(self.nodes[1].overlay.bandwidth_wallet.get_bandwidth_tokens() > 0)
  File "/Library/Python/2.7/site-packages/twisted/trial/_synctest.py", line 395, in assertTrue
    super(_Assertions, self).assertTrue(condition, msg)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 424, in assertTrue
    raise self.failureException(msg)
FailTest: False is not true
-------------------- >> begin captured logging << --------------------
TriblerTunnelCommunity: DEBUG: Launching TriblerTunnelCommunity with prefix 0002ee588f7a0c22bafd7d293809637a3efbbaeaaa9f.
TriblerTunnelCommunity: INFO: TunnelCommunity: setting become_exitnode = False
TriblerTunnelCommunity: ERROR: Could not retrieve backup exitnode cache, file does not exist!
TrustChainCommunity: DEBUG: Launching TrustChainCommunity with prefix 00025ad767b05ae592a02488272ca2a86b847d4562e1.
TrustChainDB: DEBUG: loading database [:memory:]
TrustChainDB: DEBUG: TrustChain database path: :memory:
TrustChainDB: DEBUG: open database [:memory:]
TrustChainDB: DEBUG: PRAGMA page_size = 8192 (previously: 4096) [:memory:]
TrustChainDB: DEBUG: PRAGMA journal_mode = MEMORY (no change) [:memory:]
TrustChainDB: DEBUG: PRAGMA synchronous = NORMAL (previously: 2) [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: The trustchain community started with Public Key: 4c69624e61434c504b3aa6f46fd72126da8f23c60935f611a41a5fb1767f43f157c36b846efcdaa1ec360c12781b2062420a5c364edac13f548cb0aacb27499c8daf890901c51d380d03
TriblerTunnelCommunity: DEBUG: Launching TriblerTunnelCommunity with prefix 0002ee588f7a0c22bafd7d293809637a3efbbaeaaa9f.
TriblerTunnelCommunity: INFO: TunnelCommunity: setting become_exitnode = False
TriblerTunnelCommunity: ERROR: Could not retrieve backup exitnode cache, file does not exist!
TrustChainCommunity: DEBUG: Launching TrustChainCommunity with prefix 00025ad767b05ae592a02488272ca2a86b847d4562e1.
TrustChainDB: DEBUG: loading database [:memory:]
TrustChainDB: DEBUG: TrustChain database path: :memory:
TrustChainDB: DEBUG: open database [:memory:]
TrustChainDB: DEBUG: PRAGMA page_size = 8192 (previously: 4096) [:memory:]
TrustChainDB: DEBUG: PRAGMA journal_mode = MEMORY (no change) [:memory:]
TrustChainDB: DEBUG: PRAGMA synchronous = NORMAL (previously: 2) [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: The trustchain community started with Public Key: 4c69624e61434c504b3a1482c6e5e7ed7625992f3744e500ea83043a2c5d80b54d8565e782ed4bb60d73b73b12a1ddc1a094faf3bb2a972c89f0116ccc92fbba1614f482689912992c8c
TriblerTunnelCommunity: DEBUG: Launching TriblerTunnelCommunity with prefix 0002ee588f7a0c22bafd7d293809637a3efbbaeaaa9f.
TriblerTunnelCommunity: INFO: TunnelCommunity: setting become_exitnode = False
TriblerTunnelCommunity: ERROR: Could not retrieve backup exitnode cache, file does not exist!
TrustChainCommunity: DEBUG: Launching TrustChainCommunity with prefix 00025ad767b05ae592a02488272ca2a86b847d4562e1.
TrustChainDB: DEBUG: loading database [:memory:]
TrustChainDB: DEBUG: TrustChain database path: :memory:
TrustChainDB: DEBUG: open database [:memory:]
TrustChainDB: DEBUG: PRAGMA page_size = 8192 (previously: 4096) [:memory:]
TrustChainDB: DEBUG: PRAGMA journal_mode = MEMORY (no change) [:memory:]
TrustChainDB: DEBUG: PRAGMA synchronous = NORMAL (previously: 2) [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: The trustchain community started with Public Key: 4c69624e61434c504b3af6ef8dc476bb81494b4e805f3b58748533c8869d21fe373437e88fce771b0c3d3fe1c238a6ce7a913edaca5be8533601ae1d5d005807e0aa7b6bbf8ee915edd3
TriblerTunnelCommunity: INFO: Creating 1 introduction point(s)
TriblerTunnelCommunity: INFO: Creating a new circuit of length 2 (type: IP)
TriblerTunnelCommunity: INFO: Look for a first hop that is not an exit node and is not used before
RequestCache: DEBUG: add <CircuitRequestCache anon-circuit-1253003685>
TriblerTunnelCommunity: INFO: Creating circuit 1253003685 of 2 hops. First hop: 11.164.166.117:50379
TriblerTunnelCommunity: DEBUG: Got create (1253003685) from ('80.59.198.90', 49725), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: INFO: TunnelCommunity: we joined circuit 1253003685 with neighbour ('80.59.198.90', 49725)
RequestCache: DEBUG: add <CreatedRequestCache anon-created-1253003685>
TriblerTunnelCommunity: DEBUG: Got created (1253003685) from ('11.164.166.117', 50379), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: INFO: Extending circuit 1253003685 with 4c69624e61434c504b3aa6f46fd72126da8f23c60935f611a41a5fb1767f43f157c36b846efcdaa1ec360c12781b2062420a5c364edac13f548cb0aacb27499c8daf890901c51d380d03
TriblerTunnelCommunity: DEBUG: Got extend (1253003685) from ('80.59.198.90', 49725), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: INFO: On_extend send CREATE for circuit (('80.59.198.90', 49725), 1253003685) to 119.166.244.189:3446
TriblerTunnelCommunity: INFO: Extending circuit, got candidate with IP 119.166.244.189:3446 from cache
RequestCache: DEBUG: add <ExtendRequestCache anon-circuit-138380028>
TriblerTunnelCommunity: DEBUG: Got create (138380028) from ('11.164.166.117', 50379), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: INFO: TunnelCommunity: we joined circuit 138380028 with neighbour ('11.164.166.117', 50379)
RequestCache: DEBUG: add <CreatedRequestCache anon-created-138380028>
TriblerTunnelCommunity: DEBUG: Got created (138380028) from ('119.166.244.189', 3446), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: INFO: Got CREATED message forward as EXTENDED to origin.
TriblerTunnelCommunity: DEBUG: Got extended (1253003685) from ('11.164.166.117', 50379), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
RequestCache: DEBUG: add <IPRequestCache establish-intro-41542>
TriblerTunnelCommunity: INFO: Established introduction tunnel 1253003685
TriblerTunnelCommunity: DEBUG: Got establish-intro (1253003685) from ('80.59.198.90', 49725), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: DEBUG: Relaying establish-intro from 1253003685 to 138380028
TriblerTunnelCommunity: DEBUG: Got establish-intro (138380028) from ('11.164.166.117', 50379), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got intro-established (138380028) from ('119.166.244.189', 3446), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: DEBUG: Relaying intro-established from 138380028 to 1253003685
TriblerTunnelCommunity: DEBUG: Got intro-established (1253003685) from ('11.164.166.117', 50379), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: INFO: Got intro-established from ('11.164.166.117', 50379)
TriblerTunnelCommunity: DEBUG: Launching TriblerTunnelCommunity with prefix 0002ee588f7a0c22bafd7d293809637a3efbbaeaaa9f.
TriblerTunnelCommunity: INFO: TunnelCommunity: setting become_exitnode = False
TriblerTunnelCommunity: ERROR: Could not retrieve backup exitnode cache, file does not exist!
TrustChainCommunity: DEBUG: Launching TrustChainCommunity with prefix 00025ad767b05ae592a02488272ca2a86b847d4562e1.
TrustChainDB: DEBUG: loading database [:memory:]
TrustChainDB: DEBUG: TrustChain database path: :memory:
TrustChainDB: DEBUG: open database [:memory:]
TrustChainDB: DEBUG: PRAGMA page_size = 8192 (previously: 4096) [:memory:]
TrustChainDB: DEBUG: PRAGMA journal_mode = MEMORY (no change) [:memory:]
TrustChainDB: DEBUG: PRAGMA synchronous = NORMAL (previously: 2) [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: The trustchain community started with Public Key: 4c69624e61434c504b3a6840b3fc658f03c60d3f19a7d4f4d67375da14687b07f6e7d2f0cefecfe3221051758eb8bb40ca0e2ad00fbd6b17bc5a3599da57433e69f0c9421b3d94fdce7c
TriblerTunnelCommunity: INFO: Want 1 data circuits of length 1
TriblerTunnelCommunity: INFO: Creating a new circuit of length 1 (type: DATA)
TriblerTunnelCommunity: INFO: First hop is required exit
RequestCache: DEBUG: add <CircuitRequestCache anon-circuit-2621025437>
TriblerTunnelCommunity: INFO: Creating circuit 2621025437 of 1 hops. First hop: 108.205.205.62:18886
TriblerTunnelCommunity: INFO: Removed candidate from exit_candidates dictionary
TriblerTunnelCommunity: DEBUG: Got create (2621025437) from ('119.166.244.189', 3446), I am Peer<108.205.205.62:18886, hJUqLGTKQmfkzrT0QT5yqkckqyA=>
TriblerTunnelCommunity: INFO: TunnelCommunity: we joined circuit 2621025437 with neighbour ('119.166.244.189', 3446)
RequestCache: DEBUG: add <CreatedRequestCache anon-created-2621025437>
TriblerTunnelCommunity: DEBUG: Got created (2621025437) from ('108.205.205.62', 18886), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: INFO: Do DHT request: select circuit
TriblerTunnelCommunity: INFO: Do DHT request: send dht request
RequestCache: DEBUG: add <DHTRequestCache dht-request-20276>
TriblerTunnelCommunity: DEBUG: Got dht-request (2621025437) from ('119.166.244.189', 3446), I am Peer<108.205.205.62:18886, hJUqLGTKQmfkzrT0QT5yqkckqyA=>
TriblerTunnelCommunity: INFO: Doing dht hidden seeders lookup for info_hash 80ba560466b981c0dfee282a28743519fc2fc61c
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('108.205.205.62', 18886), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('108.205.205.62', 18886)
TriblerTunnelCommunity: DEBUG: Giving incoming data packet to IPv8 (circuit ID 2621025437)
TriblerTunnelCommunity: INFO: Received dht response containing 1 peers
TriblerTunnelCommunity: INFO: Requesting key from dht peer ('119.166.244.189', 3446)
TriblerTunnelCommunity: INFO: Create key request: select circuit
TriblerTunnelCommunity: INFO: Create key request: send key request
RequestCache: DEBUG: add <KeyRequestCache key-request-18306>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('119.166.244.189', 3446), I am Peer<108.205.205.62:18886, hJUqLGTKQmfkzrT0QT5yqkckqyA=>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('119.166.244.189', 3446)
TriblerTunnelCommunity: DEBUG: Data for circuit 2621025437 exiting tunnel (('119.166.244.189', 3446))
TriblerTunnelCommunity: INFO: On key request: relay key request
RequestCache: DEBUG: add <KeyRelayCache key-request-17606>
TriblerTunnelCommunity: DEBUG: Got data (138380028) from ('119.166.244.189', 3446), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: DEBUG: Relaying data from 138380028 to 1253003685
TriblerTunnelCommunity: DEBUG: Got data (1253003685) from ('11.164.166.117', 50379), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: DEBUG: Got data (1253003685) from ('11.164.166.117', 50379)
TriblerTunnelCommunity: DEBUG: Giving incoming data packet to IPv8 (circuit ID 1253003685)
TriblerTunnelCommunity: INFO: On key request: respond with keys to ('119.166.244.189', 3446)
TriblerTunnelCommunity: DEBUG: Got data (1253003685) from ('80.59.198.90', 49725), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: DEBUG: Relaying data from 1253003685 to 138380028
TriblerTunnelCommunity: DEBUG: Got data (138380028) from ('11.164.166.117', 50379), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got data (138380028) from ('11.164.166.117', 50379)
TriblerTunnelCommunity: DEBUG: Data for circuit 138380028 exiting tunnel (('119.166.244.189', 3446))
TriblerTunnelCommunity: INFO: On key response: forward message because received over socket
MockTunnelExitSocket: DEBUG: Tunnel data to origin ('0.0.0.0', 0) for circuit 2621025437
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('108.205.205.62', 18886), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('108.205.205.62', 18886)
TriblerTunnelCommunity: DEBUG: Giving incoming data packet to IPv8 (circuit ID 2621025437)
TriblerTunnelCommunity: INFO: On key response: received keys
TriblerTunnelCommunity: INFO: Added key to peer exchange cache
TriblerTunnelCommunity: INFO: Create end-to-end on pex_peer ('119.166.244.189', 3446)
TriblerTunnelCommunity: INFO: Create end to end initiated here
RequestCache: DEBUG: add <E2ERequestCache e2e-request-1814>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('119.166.244.189', 3446), I am Peer<108.205.205.62:18886, hJUqLGTKQmfkzrT0QT5yqkckqyA=>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('119.166.244.189', 3446)
TriblerTunnelCommunity: DEBUG: Data for circuit 2621025437 exiting tunnel (('119.166.244.189', 3446))
TriblerTunnelCommunity: INFO: On create e2e: forward message because received over socket
TriblerTunnelCommunity: DEBUG: Got data (138380028) from ('119.166.244.189', 3446), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: DEBUG: Relaying data from 138380028 to 1253003685
TriblerTunnelCommunity: DEBUG: Got data (1253003685) from ('11.164.166.117', 50379), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: DEBUG: Got data (1253003685) from ('11.164.166.117', 50379)
TriblerTunnelCommunity: DEBUG: Giving incoming data packet to IPv8 (circuit ID 1253003685)
TriblerTunnelCommunity: INFO: On create e2e: create rendezvous point
TriblerTunnelCommunity: INFO: Creating a new circuit of length 1 (type: RP)
TriblerTunnelCommunity: INFO: First hop is required exit
RequestCache: DEBUG: add <CircuitRequestCache anon-circuit-10177680>
TriblerTunnelCommunity: INFO: Creating circuit 10177680 of 1 hops. First hop: 119.166.244.189:3446
TriblerTunnelCommunity: DEBUG: Got create (10177680) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: INFO: TunnelCommunity: we joined circuit 10177680 with neighbour ('80.59.198.90', 49725)
RequestCache: DEBUG: add <CreatedRequestCache anon-created-10177680>
TriblerTunnelCommunity: DEBUG: Got created (10177680) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
RequestCache: DEBUG: add <RPRequestCache establish-rendezvous-48515>
TriblerTunnelCommunity: DEBUG: Got establish-rendezvous (10177680) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got rendezvous-established (10177680) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: DEBUG: Got data (1253003685) from ('80.59.198.90', 49725), I am Peer<11.164.166.117:50379, aac6l362wFZVaUKHOthF/1X+lUU=>
TriblerTunnelCommunity: DEBUG: Relaying data from 1253003685 to 138380028
TriblerTunnelCommunity: DEBUG: Got data (138380028) from ('11.164.166.117', 50379), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got data (138380028) from ('11.164.166.117', 50379)
TriblerTunnelCommunity: DEBUG: Data for circuit 138380028 exiting tunnel (('43.239.192.252', 24382))
MockTunnelExitSocket: DEBUG: Tunnel data to origin ('0.0.0.0', 0) for circuit 2621025437
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('108.205.205.62', 18886), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got data (2621025437) from ('108.205.205.62', 18886)
TriblerTunnelCommunity: DEBUG: Giving incoming data packet to IPv8 (circuit ID 2621025437)
TriblerTunnelCommunity: INFO: Creating a new circuit of length 2 (type: RENDEZVOUS)
TriblerTunnelCommunity: INFO: Look for a first hop that is not an exit node and is not used before
RequestCache: DEBUG: add <CircuitRequestCache anon-circuit-883211545>
TriblerTunnelCommunity: INFO: Creating circuit 883211545 of 2 hops. First hop: 80.59.198.90:49725
TriblerTunnelCommunity: DEBUG: Got create (883211545) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: INFO: TunnelCommunity: we joined circuit 883211545 with neighbour ('119.166.244.189', 3446)
RequestCache: DEBUG: add <CreatedRequestCache anon-created-883211545>
TriblerTunnelCommunity: DEBUG: Got created (883211545) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: INFO: Extending circuit 883211545 with 4c69624e61434c504b3aa6f46fd72126da8f23c60935f611a41a5fb1767f43f157c36b846efcdaa1ec360c12781b2062420a5c364edac13f548cb0aacb27499c8daf890901c51d380d03
TriblerTunnelCommunity: DEBUG: Got extend (883211545) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: INFO: On_extend send CREATE for circuit (('119.166.244.189', 3446), 883211545) to 119.166.244.189:3446
TriblerTunnelCommunity: INFO: Extending circuit, got candidate with IP 119.166.244.189:3446 from cache
RequestCache: DEBUG: add <ExtendRequestCache anon-circuit-995213262>
TriblerTunnelCommunity: DEBUG: Got create (995213262) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: INFO: TunnelCommunity: we joined circuit 995213262 with neighbour ('80.59.198.90', 49725)
RequestCache: DEBUG: add <CreatedRequestCache anon-created-995213262>
TriblerTunnelCommunity: DEBUG: Got created (995213262) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: INFO: Got CREATED message forward as EXTENDED to origin.
TriblerTunnelCommunity: DEBUG: Got extended (883211545) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
RequestCache: DEBUG: add <LinkRequestCache link-request-23169>
TriblerTunnelCommunity: DEBUG: Got link-e2e (883211545) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: DEBUG: Relaying link-e2e from 883211545 to 995213262
TriblerTunnelCommunity: DEBUG: Got link-e2e (995213262) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: DEBUG: Got linked-e2e (995213262) from ('119.166.244.189', 3446), I am Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=>
TriblerTunnelCommunity: DEBUG: Relaying linked-e2e from 995213262 to 883211545
TriblerTunnelCommunity: DEBUG: Got linked-e2e (883211545) from ('80.59.198.90', 49725), I am Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=>
TriblerTunnelCommunity: ERROR: On linked e2e: could not find download!
TriblerTunnelCommunity: ERROR: On linked e2e: could not find download for ??Vf?????(*(t5?/?!
TriblerTunnelCommunity: INFO: Sending payout of 1310720000 (base: 262144000) to Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=> (cid: 883211545)
TrustChainDB: DEBUG: commit [:memory:]
TriblerTunnelCommunity: INFO: Removing circuit 883211545 
TriblerTunnelCommunity: INFO: destroy_circuit 883211545 ('80.59.198.90', 49725)
TriblerTunnelCommunity: INFO: Got destroy from ('119.166.244.189', 3446) for circuit 883211545
TriblerTunnelCommunity: INFO: Found relay 883211545 -> 995213262 (('119.166.244.189', 3446))
TriblerTunnelCommunity: INFO: Fw destroy to 995213262 ('119.166.244.189', 3446)
TriblerTunnelCommunity: INFO: Found relay 995213262 -> 883211545 (('119.166.244.189', 3446))
TriblerTunnelCommunity: INFO: Got destroy from ('80.59.198.90', 49725) for circuit 995213262
TriblerTunnelCommunity: INFO: Found relay 10177680 -> 995213262 (('80.59.198.90', 49725))
TriblerTunnelCommunity: INFO: Found relay 995213262 -> 10177680 (('80.59.198.90', 49725))
TriblerTunnelCommunity: INFO: Fw destroy to 10177680 ('80.59.198.90', 49725)
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function partial_next at 0x10b94db18>, [], (Block 768423e2 from ...1d380d03:1 links ...e915edd3:0 for {'down': 1310720000, 'total_down': 1310720000, 'up': 0, 'total_up': 0} type tribler_bandwidth)
TriblerTunnelCommunity: INFO: Got destroy from ('119.166.244.189', 3446) for circuit 10177680
TriblerTunnelCommunity: INFO: Got a circuit 10177680 ('119.166.244.189', 3446)
TrustChainCommunity: INFO: Received request block addressed to us (Block 768423e2 from ...1d380d03:1 links ...e915edd3:0 for {'down': 1310720000, 'total_down': 1310720000, 'up': 0, 'total_up': 0} type tribler_bandwidth)
TriblerTunnelCommunity: INFO: Removing circuit 10177680 Got destroy
TriblerTunnelCommunity: INFO: Removed rendezvous point 883211545
TrustChainCommunity: INFO: Signed block to 1d380d03 (Block 4078bb23 from ...e915edd3:1 links ...1d380d03:1 for {'down': 0, 'total_down': 0, 'up': 1310720000, 'total_up': 1310720000} type tribler_bandwidth) validation result (<function partial_next at 0x10b94db18>, [])
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: Sending block to (119.166.244.189:3446) (Block 4078bb23 from ...e915edd3:1 links ...1d380d03:1 for {'down': 0, 'total_down': 0, 'up': 1310720000, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: DEBUG: Broadcasting blocks Block 768423e2 from ...1d380d03:1 links ...e915edd3:0 for {'down': 1310720000, 'total_down': 1310720000, 'up': 0, 'total_up': 0} type tribler_bandwidth and Block 4078bb23 from ...e915edd3:1 links ...1d380d03:1 for {'down': 0, 'total_down': 0, 'up': 1310720000, 'total_up': 1310720000} type tribler_bandwidth
TriblerTunnelCommunity: INFO: Sending payout of 786432000 (base: 262144000) to Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=> (cid: 995213262)
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function no_info at 0x10b94dc08>, [], (Block e3a95ee2 from ...e915edd3:2 links ...1d380d03:0 for {'down': 786432000, 'total_down': 786432000, 'up': 0, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: INFO: Received request block addressed to us (Block e3a95ee2 from ...e915edd3:2 links ...1d380d03:0 for {'down': 786432000, 'total_down': 786432000, 'up': 0, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: INFO: Request block could not be validated sufficiently, crawling requester. (<function no_info at 0x10b94dc08>, [])
RequestCache: DEBUG: add <CrawlRequestCache crawl-7608034>
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Requesting crawl of node e915edd3 (blocks 1 to 1) with id 7608034
TriblerTunnelCommunity: INFO: Sending payout of 262144000 (base: 262144000) to Peer<80.59.198.90:49725, 41YSD7cbow62kOLCU+J9wGjKX6A=> (cid: 10177680)
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function valid at 0x10b94da28>, [], (Block 4078bb23 from ...e915edd3:1 links ...1d380d03:1 for {'down': 0, 'total_down': 0, 'up': 1310720000, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function partial_next at 0x10b94db18>, [], (Block 8142660a from ...1d380d03:2 links ...e915edd3:0 for {'down': 262144000, 'total_down': 1572864000, 'up': 0, 'total_up': 0} type tribler_bandwidth)
TrustChainCommunity: INFO: Received request block addressed to us (Block 8142660a from ...1d380d03:2 links ...e915edd3:0 for {'down': 262144000, 'total_down': 1572864000, 'up': 0, 'total_up': 0} type tribler_bandwidth)
TrustChainCommunity: INFO: Signed block to 1d380d03 (Block d68cbe3f from ...e915edd3:3 links ...1d380d03:2 for {'down': 0, 'total_down': 786432000, 'up': 262144000, 'total_up': 1572864000} type tribler_bandwidth) validation result (<function partial_next at 0x10b94db18>, [])
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: Sending block to (119.166.244.189:3446) (Block d68cbe3f from ...e915edd3:3 links ...1d380d03:2 for {'down': 0, 'total_down': 786432000, 'up': 262144000, 'total_up': 1572864000} type tribler_bandwidth)
TrustChainCommunity: DEBUG: Broadcasting blocks Block 8142660a from ...1d380d03:2 links ...e915edd3:0 for {'down': 262144000, 'total_down': 1572864000, 'up': 0, 'total_up': 0} type tribler_bandwidth and Block d68cbe3f from ...e915edd3:3 links ...1d380d03:2 for {'down': 0, 'total_down': 786432000, 'up': 262144000, 'total_up': 1572864000} type tribler_bandwidth
TrustChainCommunity: INFO: Received crawl request from node 1d380d03 for range 1-1
TrustChainCommunity: DEBUG: Sending block for crawl request to Peer<119.166.244.189:3446, YGfirAjyNMTZIUSI2QsWeXwdGYE=> (Block 4078bb23 from ...e915edd3:1 links ...1d380d03:1 for {'down': 0, 'total_down': 0, 'up': 1310720000, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: INFO: Sent 1 blocks
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function partial_next at 0x10b94db18>, [], (Block d68cbe3f from ...e915edd3:3 links ...1d380d03:2 for {'down': 0, 'total_down': 786432000, 'up': 262144000, 'total_up': 1572864000} type tribler_bandwidth)
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function valid at 0x10b94da28>, [], (Block 4078bb23 from ...e915edd3:1 links ...1d380d03:1 for {'down': 0, 'total_down': 0, 'up': 1310720000, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: INFO: Block validation result <function valid at 0x10b94da28>, [], (Block e3a95ee2 from ...e915edd3:2 links ...1d380d03:0 for {'down': 786432000, 'total_down': 786432000, 'up': 0, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: INFO: Received request block addressed to us (Block e3a95ee2 from ...e915edd3:2 links ...1d380d03:0 for {'down': 786432000, 'total_down': 786432000, 'up': 0, 'total_up': 1310720000} type tribler_bandwidth)
TrustChainCommunity: INFO: Signed block to e915edd3 (Block 9b35edb1 from ...1d380d03:3 links ...e915edd3:2 for {'down': 0, 'total_down': 1572864000, 'up': 786432000, 'total_up': 786432000} type tribler_bandwidth) validation result (<function partial_next at 0x10b94db18>, [])
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: DEBUG: Sending block to (80.59.198.90:49725) (Block 9b35edb1 from ...1d380d03:3 links ...e915edd3:2 for {'down': 0, 'total_down': 1572864000, 'up': 786432000, 'total_up': 786432000} type tribler_bandwidth)
TrustChainCommunity: DEBUG: Broadcasting blocks Block e3a95ee2 from ...e915edd3:2 links ...1d380d03:0 for {'down': 786432000, 'total_down': 786432000, 'up': 0, 'total_up': 1310720000} type tribler_bandwidth and Block 9b35edb1 from ...1d380d03:3 links ...e915edd3:2 for {'down': 0, 'total_down': 1572864000, 'up': 786432000, 'total_up': 786432000} type tribler_bandwidth
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
TrustChainCommunity: INFO: Block validation result <function partial_next at 0x10b94db18>, [], (Block 9b35edb1 from ...1d380d03:3 links ...e915edd3:2 for {'down': 0, 'total_down': 1572864000, 'up': 786432000, 'total_up': 786432000} type tribler_bandwidth)
TrustChainDB: DEBUG: commit [:memory:]
TrustChainDB: DEBUG: commit [:memory:]
--------------------- >> end captured logging << ---------------------
```

**Comment by devos50** [comment]
It seems to be caused by [this commit](https://github.com/Tribler/py-ipv8/commit/72b9f88a9a96fa8acdd2e838cba69d57afe867fb).

**Comment by devos50** [comment]
Fixed by #4220
